From f55288cefb2e81d815b5b2469e5890581720e92f Mon Sep 17 00:00:00 2001
From: Gosha <gosha>
Date: Fri, 25 Apr 2014 17:41:56 +0400
Subject: [PATCH] ssdp location ip changed to 192.168.88.1

---
 minissdp.c | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/minissdp.c b/minissdp.c
index 4eb90a5..ca9afa3 100644
--- a/minissdp.c
+++ b/minissdp.c
@@ -239,7 +239,7 @@ SendSSDPResponse(int s, struct sockaddr_in sockname, int st_no,
 		"USN: %s%s%s%s\r\n"
 		"EXT:\r\n"
 		"SERVER: " MINIDLNA_SERVER_STRING "\r\n"
-		"LOCATION: http://%s:%u" ROOTDESC_PATH "\r\n"
+		"LOCATION: http://192.168.88.1:%u" ROOTDESC_PATH "\r\n"
 		"Content-Length: 0\r\n"
 		"\r\n",
 		(runtime_vars.notify_interval<<1)+10,
@@ -250,7 +250,7 @@ SendSSDPResponse(int s, struct sockaddr_in sockname, int st_no,
 		(st_no > 0 ? "::" : ""),
 		(st_no > 0 ? known_service_types[st_no] : ""),
 		(st_no > 1 ? "1" : ""),
-		host, (unsigned int)port);
+		(unsigned int)port);
 	DPRINTF(E_DEBUG, L_SSDP, "Sending M-SEARCH response to %s:%d ST: %s\n",
 		inet_ntoa(sockname.sin_addr), ntohs(sockname.sin_port),
 		known_service_types[st_no]);
-- 
1.8.3.msysgit.0

