diff -ur OpenRA-OpenRA-d9fbee6/OpenRA.Game/Game.cs OpenRA-OpenRA-d9fbee6-new/OpenRA.Game/Game.cs
--- OpenRA-OpenRA-d9fbee6/OpenRA.Game/Game.cs	2011-10-23 09:44:23.000000000 +0400
+++ OpenRA-OpenRA-d9fbee6-new/OpenRA.Game/Game.cs	2011-10-23 13:40:37.491793713 +0400
@@ -325,8 +325,34 @@
 
 		internal static void Run()
 		{
+			ushort mSecToWaitInCycle = (ushort) (1000/Settings.Graphics.FrameRate);
+			ushort i=0;
+			
 			while (!quit)
+			{
+				var sw = System.Diagnostics.Stopwatch.StartNew();
 				Tick( orderManager, viewport );
+				if ( Settings.Graphics.CapFrameRate && (sw.Elapsed.TotalMilliseconds < mSecToWaitInCycle ) )
+					//? Find a better function then use the following string: ?
+					//System.Threading.Thread.Sleep((int)(mSecToWaitInCycle - sw.Elapsed.TotalMilliseconds));
+				{
+					while (true)
+					{
+						if (sw.Elapsed.TotalMilliseconds >= mSecToWaitInCycle)
+							break;
+						System.Threading.Thread.Sleep(1);
+					}
+				}
+				if (Settings.Graphics.DebugFrameRate)
+				{
+					i++;
+					if (i == Settings.Graphics.FrameRate / 2)
+					{
+						Debug("Frame Rate: {0}", (float) 1000.0/sw.Elapsed.TotalMilliseconds);
+						i = 0;
+					}
+				}
+			}
 
 			OnQuit();
 		}
diff -ur OpenRA-OpenRA-d9fbee6/OpenRA.Game/GameRules/Settings.cs OpenRA-OpenRA-d9fbee6-new/OpenRA.Game/GameRules/Settings.cs
--- OpenRA-OpenRA-d9fbee6/OpenRA.Game/GameRules/Settings.cs	2011-10-23 09:44:23.000000000 +0400
+++ OpenRA-OpenRA-d9fbee6-new/OpenRA.Game/GameRules/Settings.cs	2011-10-23 13:42:30.973736335 +0400
@@ -60,6 +60,9 @@
 		public int2 FullscreenSize = new int2(0,0);
 		public int2 WindowedSize = new int2(1024, 768);
 		public bool PixelDouble = false;
+		public bool CapFrameRate = false;
+		public bool DebugFrameRate = false;
+		public ushort FrameRate = 70;
 
 		public int BatchSize = 8192;
 		public int NumTempBuffers = 8;
