--- 1/Makefile	2014-05-25 16:10:57.000000000 +0400
+++ 2/Makefile	2014-05-28 12:56:04.706132467 +0400
@@ -327,7 +327,6 @@
 	@$(INSTALL_PROGRAM) thirdparty/MaxMind.GeoIP2.dll "$(DATA_INSTALL_DIR)"
 	@$(INSTALL_PROGRAM) thirdparty/Newtonsoft.Json.dll "$(DATA_INSTALL_DIR)"
 	@$(INSTALL_PROGRAM) thirdparty/RestSharp.dll "$(DATA_INSTALL_DIR)"
-	@$(CP) thirdparty/${platformdeps}/* "$(DATA_INSTALL_DIR)"
 
 ifeq ($(shell uname),Linux)
 	@$(CP) *.sh "$(DATA_INSTALL_DIR)"
