--- 1/Makefile	2013-12-02 08:42:59.073593359 +0400
+++ 2/Makefile	2013-12-03 00:39:49.791997757 +0400
@@ -48,10 +48,10 @@
 datarootdir ?= $(prefix)/share
 datadir ?= $(datarootdir)
 bindir ?= $(prefix)/bin
-libexecdir ?= $(prefix)/lib
+libdir ?= $(prefix)/lib
 BIN_INSTALL_DIR = $(DESTDIR)$(bindir)
 # TODO: separate data and binaries properly
-DATA_INSTALL_DIR = $(DESTDIR)$(libexecdir)/openra
+DATA_INSTALL_DIR = $(DESTDIR)$(datadir)/openra
 
 # install tools
 RM = rm
@@ -349,14 +349,15 @@
 	@$(CP) thirdparty/SharpFont.dll.config "$(DATA_INSTALL_DIR)"
 	@$(INSTALL_PROGRAM) thirdparty/Mono.Nat.dll "$(DATA_INSTALL_DIR)"
 	@$(CP) thirdparty/LuaInterface.dll.config "$(DATA_INSTALL_DIR)"
-	@$(INSTALL_PROGRAM) thirdparty/liblua-linux32.so "$(DATA_INSTALL_DIR)"
-	@$(INSTALL_PROGRAM) thirdparty/liblua-linux64.so "$(DATA_INSTALL_DIR)"
-	@$(INSTALL_PROGRAM) thirdparty/liblua-osx.dylib "$(DATA_INSTALL_DIR)"
+	@case "$${OSTYPE}" in \
+		linux*)   $(INSTALL_PROGRAM) thirdparty/liblua-linux$$(getconf LONG_BIT).so "$(DATA_INSTALL_DIR)"  ;; \
+		darwin*) $(INSTALL_PROGRAM) thirdparty/liblua-osx.dylib "$(DATA_INSTALL_DIR)"  ;; \
+	esac
 
 	@echo "#!/bin/sh" 				>  openra
 	@echo 'BINDIR=$$(dirname $$(readlink -f $$0))'	>> openra
 	@echo 'ROOTDIR="$${BINDIR%'"$(bindir)"'}"' 	>> openra
-	@echo 'EXECDIR="$${ROOTDIR}'"$(libexecdir)"'"'	>> openra
+	@echo 'EXECDIR="$${ROOTDIR}'"$(datadir)"'"'	>> openra
 	@echo 'cd "$${EXECDIR}/openra"' 		>> openra
 	@echo 'exec mono OpenRA.Game.exe "$$@"' 	>> openra
 	@$(INSTALL_DIR) "$(BIN_INSTALL_DIR)"
@@ -371,7 +372,7 @@
 	@echo "#!/bin/sh" 				>  openra-editor
 	@echo 'BINDIR=$$(dirname $$(readlink -f $$0))'	>> openra-editor
 	@echo 'ROOTDIR="$${BINDIR%'"$(bindir)"'}"' 	>> openra-editor
-	@echo 'EXECDIR="$${ROOTDIR}'"$(libexecdir)"'"'	>> openra-editor
+	@echo 'EXECDIR="$${ROOTDIR}'"$(datadir)"'"'	>> openra-editor
 	@echo 'cd "$${EXECDIR}/openra"'			>> openra-editor
 	@echo 'exec mono OpenRA.Editor.exe "$$@"'	>> openra-editor
 	@$(INSTALL_DIR) "$(BIN_INSTALL_DIR)"
